uniffi/src/testing.rs

```diff
@@ -37,15 +35,22 @@ lazy_static! {
 /// a foreign-language test file that exercises that component's bindings. It ensures that the
 /// component is compiled and available for use and then executes the foreign language script,
 /// returning successfully iff the script exits successfully.
-pub fn run_foreign_language_testcase<P: AsRef<Path>>(pkg_dir: &str, test_file: P) -> Result<()> {
-    let test_file: &Path = test_file.as_ref();
+pub fn run_foreign_language_testcase(pkg_dir: &str, idl_file: &str, test_file: &str) -> Result<()> {
     let cdylib_file = ensure_compiled_cdylib(pkg_dir)?;
     let out_dir = Path::new(cdylib_file.as_str())
         .parent()
-        .ok_or_else(|| anyhow::anyhow!("Generated cdylib has no parent directory"))?;
-    let lang = test_file.extension().unwrap_or_default();
-    ensure_generated_bindings(cdylib_file.as_str(), out_dir, lang)?;
-    bindings::run_script(Some(out_dir), Some(test_file), lang.try_into()?)
+        .ok_or_else(|| anyhow::anyhow!("Generated cdylib has no parent directory"))?
+        .to_str()
+        .unwrap();
+
+    let _lock = UNIFFI_BINDGEN.lock();
```

Comment:
I put all the `compile` and `run_script` machinery back in `uniffi_ bindgen`, leaving a single call to `uniffi-bindgen`. As a bonus we can now run tests outside of Rust.